src/databricks/labs/ucx/source_code/jobs.py

```diff
@@ -150,6 +151,17 @@ def _register_existing_cluster_id(self, graph: DependencyGraph):  # pylint: disa
             return
         # TODO: https://github.com/databrickslabs/ucx/issues/1637
         # load libraries installed on the referred cluster
+
+        libraries_api = compute.LibrariesAPI(self._ws.api_client)
+        library_full_status_list = libraries_api.cluster_status(self._task.existing_cluster_id)
+
+        libraries_from_cluster = []
+        # TODO: Lint the libraries obtained from the cluster
+        for library_full_status in library_full_status_list:
+            if library_full_status.library:
+                if library_full_status.library.pypi:
+                    libraries_from_cluster.append(library_full_status.library.pypi.package)
```

Comment:
I think we can reuse the `_lint_library

```suggestion
                    yield self._lint_library(library_full_status.library, graph)
```